packages/vite/src/node/build.ts

```diff
@@ -959,7 +960,7 @@ function injectSsrFlagToHooks(plugin: Plugin): Plugin {
 function wrapSsrResolveId(hook?: Plugin['resolveId']): Plugin['resolveId'] {
   if (!hook) return
 
-  const fn = 'handler' in hook ? hook.handler : hook
+  const fn = getHookHandler(hook)
```

Comment:
I was also thinking if we should handle if `fn` is `undefined` here, but I suppose the types always restricts that you need a `handler` with the object-form, so I suppose this is also fine for now.